spec/requests/api/v1/user_event_request_spec.rb

```diff
@@ -28,25 +30,50 @@
     end
 
     it "can create a new user event with necessary keys" do
-        headers = { "CONTENT_TYPE" => "application/json" }
 
-        post "/api/v1/user_events", params: '{"event" : {"name" : "test button", "event_type" : "click",    "at" : "2020-06-12T00:00:01", "button_color" : "red" }}', headers: headers
+        post "/api/v1/user_events", params: '{"event" : {"name" : "test button", "event_type" : "click",    "at" : "2020-06-12T00:00:01", "button_color" : "red" }}', headers: @header
 
         event = UserEvent.last
 
         expect(response).to be_successful
+        expect(response).to have_http_status(200)
         expect(UserEvent.count).to eq(1)
         expect(event.event_type).to eq('click')
     end
 
-    it "cannot create new user event without necessary keys" do
-        headers = { "CONTENT_TYPE" => "application/json" }
+    it "cannot create new user event without necessary name key" do
 
-        post "/api/v1/user_events", params: '{"event" : {"wrong_key" : "test button", "event_type" : "click",    "at" : "2020-06-12T00:00:01", "button_color" : "red" }}', headers: headers
+        post "/api/v1/user_events", params: '{"event" : {"wrong_key" : "test button", "event_type" : "click",    "at" : "2020-06-12T00:00:01", "button_color" : "red" }}', headers: @header
 
-        event = UserEvent.last
-        
-        expect(response).to be_successful
+        body = JSON.parse(response.body, symbolize_names: true)
+
```

Comment:
Tests are checking to make sure the correct http response was returned as well as correct message indicating which keys, if any, are missing.